src/shared/nss-record.c

```diff
@@ -55,18 +78,19 @@ int nss_passwd_to_user_record(
                 free_and_replace(hr->real_name, mangled);
         }
 
-        r = free_and_strdup(&hr->home_directory, empty_to_null(pwd->pw_dir));
+        r = free_and_strdup(&hr->home_directory, utf8_only(empty_to_null(pwd->pw_dir)));
```

Comment:
so i thought about instead inserting unicode char 0xFFFD wherever we encounter invalid utf8. i.e. the unicode replacement char.

i.e. pass on the data as closely as we can, but minimally replace what we can't allow with a recognizable marker. don't surpess whole strings, but just the offending chars

Reply:
So, normally we shouldn't have any invalid chars. E.g. for the password has it says that is contains only printable characters. Obviously this is not always respected as shown by the reported bugs, but that should be a fairly rare case, mostly applying to systems that were installed a long time ago and have been upgraded ever since. And if we replace with a replacement char, we'd have to add handing in various places to understand that the string is not the real string. I think it's cleaner and easier to just drop those (rare) invalid strings.